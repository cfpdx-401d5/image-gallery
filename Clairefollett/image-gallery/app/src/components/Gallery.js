import React from 'react';

export default function Gallery(props) {
    const galleryObj = props.puppies.map(puppies => {
        return (
            <GalleryDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete} />
        );
    });
    return (
        <ul>{galleryObj}</ul>
    );
}

function GalleryDisplay(props) {  
        return (
            <div className="clearfix">
                <li>
                    <p>Title: {props.puppies.title}</p>
                    <img className='gallery-img' src={props.puppies.url} alt='' />
                    <p>Description: {props.puppies.description}</p>
                    <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
                </li>
            </div>
        );
}

