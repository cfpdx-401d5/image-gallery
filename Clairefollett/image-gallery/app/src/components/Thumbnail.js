import React from 'react';

export default function Thumbnail(props) {
    const thumbnail = props.puppies.map(puppies => {
        return (
            <ThumbnailDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete}/>
        );
    });
    return (
      <ul>{thumbnail}</ul>
    );
}

function ThumbnailDisplay(props) {
  return (
    <div className="clearfix">
      <li>
        <p height='20px' width='200'>title: {props.puppies.title}</p>
        <img className='thumbnail' src={props.puppies.url} alt='' height='100' width='100' />
        <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
      </li>
    </div>
  );
}
