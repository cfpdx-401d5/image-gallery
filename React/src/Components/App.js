import React from 'react';

import CreateBunnyForm from './CreateBunnyForm';
import Selector from './Selector';
import View from './View';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      viewStyle: null,
      babyBunBuns: [],
    };
    this.handleClick = this.handleClick.bind(this);
    this.deleteBunny = this.deleteBunny.bind(this);
    this.addBunny = this.addBunny.bind(this);
  }


  handleClick(viewValue) {
    console.log(viewValue);
    this.setState({
      viewStyle: viewValue
    });
  }

  deleteBunny(babyBunIndex, _id){
    fetch(`http://localhost:8000/bunniess/${_id}`, {method: 'DELETE'})
      .then(res => res.json())
      .then(res => {
          if(res.message) {
            //res.message;
            let babyBunBunsFiltered = this.state.babyBunBuns.filter((element, i)=>{
                if (i !== babyBunIndex) {
                    return element
                }
            });
            this.setState({
                babyBunBuns: babyBunBunsFiltered
            })
        }
    })
  }

  addBunny(newBunBun){
    fetch('http://localhost:8000/bunnies', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newBunBun)
    })
      .then(res => res.json())
      .then(res => {
          delete res.__v;
          let babyBunBunsMultiplied = this.state.babyBunBuns.concat(res);
          this.setState({
              babyBunBuns: babyBunBunsMultiplied
          })
      })
  }

  render() {
    return (
      <div>
        <Selector 
          handleClick={this.handleClick}
        />
        <CreateBunnyForm 
          addBunny={this.addBunny}
        />
        <View 
          deleteBunny={this.deleteBunny} 
          babyBunBuns={this.state.babyBunBuns} 
          viewStyle={this.state.viewStyle}
        />
      </div>
    );
  }

  componentDidMount(){
    fetch('http://localhost:8000/bunnies')
      .then(res => res.json())
      .then(bunbuns => {
        this.setState(
          {babyBunBuns: bunbuns}
        )
      })
  }
}

