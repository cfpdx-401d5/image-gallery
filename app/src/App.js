import React, { Component } from 'react';

import ListDisplay from './displays/ListDisplay';
import ThumbnailDisplay from './displays/ThumbnailDisplay';
import GalleryDisplay from './displays/GalleryDisplay';
import DisplaySelector from './DisplaySelector';
import FormContainer from './image_form/FormContainer';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayType: 'list',
      images: [{
        id: 'id1',
        title: 'Cute Bunny',
        description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
        url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
      },
      {
        id: 'id2',
        title: 'Upside Down Sloth',
        description: 'Sloths just love to hang out.',
        url: 'http://kids.nationalgeographic.com/content/dam/kids/photos/animals/Mammals/Q-Z/sloth-beach-upside-down.jpg.adapt.945.1.jpg'
      },
      {
        id: 'id3',
        title: 'Waving Panda',
        description: 'When a panda waves, you wave back!',
        url: 'http://media4.s-nbcnews.com/j/newscms/2016_36/1685951/ss-160826-twip-05_8cf6d4cb83758449fd400c7c3d71aa1f.nbcnews-ux-2880-1000.jpg'
      }
      ],
    };
    this.setDisplay = this.setDisplay.bind(this);
    this.addImage = this.addImage.bind(this);
    this.deleteImage = this.deleteImage.bind(this);
  }

  setDisplay(whatDisplay) {
    this.setState({ displayType: whatDisplay });
  }

  addImage(e, imageSubmission) {
    e.preventDefault();
    // let newImgArray = [...this.state.images]; // copy the images[] in the state
    // newImgArray.push(imageSubmission); // make new image array

    // this.setState({ images: newImgArray });
    fetch('http://localhost:3001/images', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(imageSubmission)
    })
      .then(res => res.json())
      .then(res => {
        let newImgArray = [...this.state.images];
        delete res.__v;
        newImgArray.push(res);
        this.setState({ images: newImgArray });
      });
  }

  deleteImage(imageID) {
    // let newImgArray = this.state.images.filter(image => {
    //   return (image.id !== imageID);
    // });

    // this.setState({ images: newImgArray });
    fetch(`http://localhost:3001/images/${imageID}`, { method: 'DELETE' })
      .then(res => res.json())
      .then(res => {
        if (res.message) {
          let newImgArray = this.state.images.filter(image => {
            return (image.id !== imageID);
          });
          this.setState({ images: newImgArray });
        }
      });
  }

  render() {
    let DisplayType;

    if (this.state.displayType === 'list') DisplayType = ListDisplay;
    else if (this.state.displayType === 'thumbnail') DisplayType = ThumbnailDisplay;
    else if (this.state.displayType === 'gallery') DisplayType = GalleryDisplay;

    return (
        <div>
          <h1>Image Gallery</h1>
          <FormContainer addImage={this.addImage} />
          <DisplaySelector setDisplay={this.setDisplay} />
          <DisplayType images={this.state.images} deleteImage={this.deleteImage} />
        </div>
    );
  }

  componentDidMount() {
    fetch('http://localhost:3001/images')
      .then(res => res.json())
      .then(resImages => {
        this.setState({ images: resImages });
      });
  }
  
}

