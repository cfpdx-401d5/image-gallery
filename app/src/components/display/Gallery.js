import React from 'react';

export default function GalleryDisplay(props) {
    return (
      <div>
        <h1>Title: {props.resource.title}</h1>
        <img className='gallery' src={props.resource.url} alt='Cute Bunny'/>
        <p>Description: {props.resource.description}</p>
        <button onClick={() => props.onDelete(props.resource._id)}>delete</button>
      </div>
    ); 
};
