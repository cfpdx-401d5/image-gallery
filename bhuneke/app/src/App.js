import React, { Component } from 'react';
import AddResource from './components/editing/AddResource';
import GalleryDisplay from './components/display/Gallery';
import ThumbnailDisplay from './components/display/Thumbnail';
import DetailDisplay from './components/display/Detail';
import fetcher from './helpers/fetcher';

const GALLERY_TYPE = 'gallery';
const LIST_TYPE = 'list';
const THUMBNAIL_TYPE = 'thumbnail';

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedView: '',
            resources: []
        };
        this.onViewSelect = this.onViewSelect.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onAdd = this.onAdd.bind(this);
    }

    onViewSelect(e, selected) {
        e.preventDefault();
        this.setState({
            selectedView: selected
        });
    }
    doFetch() {
        fetcher({
            path: '/resources',
            method: 'GET',
        })
        .then(res => res.json())
        .then(resources => {
            this.setState({ resources });
        })
        .catch(err => { console.log(err); });
    }

    componentDidMount() {
        this.doFetch();
    }

    onDelete(id) {
        fetcher({
            path: `/resources/${id}`,
            method: 'DELETE'
        });
        this.doFetch();
    }

    onAdd(newResource) {
        let newData = this.state.resources.slice();
        newData.push(newResource);
        this.setState({
            resources: newData
        });
    }

    render() {
        return (
            <div>
                < ViewSelector onViewSelect={this.onViewSelect} />
                < ViewDisplay onDelete={this.onDelete} selectedView={this.state.selectedView} resources={this.state.resources} />
                < AddResource onAdd={this.onAdd} />
            </div>
        );
    }
}

function ViewSelector(props) {
    return (
        <div className='view-selector'>
            <button onClick={(e, selected) => props.onViewSelect(e, LIST_TYPE)}>List View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, THUMBNAIL_TYPE)}>Thumbnail View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, GALLERY_TYPE)}>Gallery View</button>
        </div>);
}

function ViewDisplay(props) {
    let selectedView = props.selectedView;
    if (selectedView) {
        return (
            <ListView
                selectedView={selectedView}
                resources={props.resources}
                onDelete={props.onDelete} 
            />);
    } else {
        return <p>Choose View!</p>;
    }
}

function ListView(props) {
    let DisplayType;
    const selectedView = props.selectedView;
    if (selectedView === LIST_TYPE) DisplayType = DetailDisplay;
    else if (selectedView === GALLERY_TYPE) DisplayType = GalleryDisplay;
    else if (selectedView === THUMBNAIL_TYPE) DisplayType = ThumbnailDisplay;

    const resourceList = props.resources.map(resource => {
        return (
            <DisplayType
                key={resource._id}
                resource={resource}
                onDelete={props.onDelete} 
            />);
    });

    return (
        <div>{resourceList}</div>
    );
}

export {
    ViewSelector,
    ViewDisplay,
    ListView,
};
