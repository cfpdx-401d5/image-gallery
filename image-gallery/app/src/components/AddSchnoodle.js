import React, { Component } from 'react';

const TITLE = 'Title';
const DESCRIPTION = 'Description';
const URL = 'Url';

export default class AddSchnoodle extends Component {
    render() {
        return(
            <div>
                <h1 className="add">Add Schnoodle:</h1>
                <form onSubmit={(e) => {
                    const newSchnoodle = {
                        id: new Date(),
                        title: this.refs.title.value,
                        url: this.refs.url.value,
                        description: this.refs.description.value,                
                    }
                    this.props.onSubmit(e, newSchnoodle)
                }}>
                    <input ref={TITLE} required placeholder={TITLE}></input>
                    <input ref={URL} required placeholder={URL}></input>
                    <input ref={DESCRIPTION} required placeholder={DESCRIPTION}></input>
                    <button className="action-button shadow animate green">Add</button>
                </form>
            </div>
        )
    }
}