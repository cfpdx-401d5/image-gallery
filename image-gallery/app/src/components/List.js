import React from 'react';

export default function List(props) {
    const listItem = props.puppies.map(puppies => {
        return (
            <ListDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete}/>
        );
    });
    return (
      <ul>{listItem}</ul>
    );
}

function ListDisplay(props) {
    return (
        <div className="clearfix">
          <li>
            <p>Title: {props.puppies.title}</p>
            <p>Description: {props.puppies.description}</p>
            <p>Url: <a href={props.puppies.url}>{props.puppies.url}</a></p>
            <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
          </li>
        </div>
    );
}

