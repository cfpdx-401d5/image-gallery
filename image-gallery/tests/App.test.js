/* eslint-disable */

import React from 'react';
import {shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/app';
import List from '../src/view/list';
import Gallery from '../src/view/gallery';
import Thumbnail from '../src/view/thumbnail'
import AddSchnoodle from '../src/forms/form';
import Display from '../src/view/display'

describe('my buttons click handlers', () => {
    it('runs my test!', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

describe('adding a new schnoodle', () => {
    it('adds a shcnoodle', () => {
        const rendered = shallow(<AddSchnoodle array= {[]} onDelete={() => {}} onChange={() => {{}}} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

describe('rendering list', () => {
    it('displays list', () => {
        const rendered = shallow(<Display array={[]} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

describe('thumbnail view', () => {
    it('thumbnail display', () => {
        const rendered = shallow(<Thumbnail array={[]} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

describe('gallery view', () => {
    it('gallery display', () => {
        const rendered = shallow(<Gallery array={[]} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

describe('list view', () => {
    it('list display', () => {
        const rendered = shallow(<List array={[]} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});


//none of my other tests will work