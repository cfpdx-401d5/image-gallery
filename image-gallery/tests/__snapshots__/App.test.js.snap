exports[`adding a new schnoodle adds a shcnoodle 1`] = `
<form>
  <fieldset>
    <legend
      className="view">
      Add your schnoodle 
    </legend>
    <div
      className="formz">
      <TextField
        content=""
        controlFunc={[Function]}
        inputType="text"
        name="title"
        placeholder="Please type the name of your schnoodle for the title"
        title="Title: " />
      <TextArea
        content=""
        controlFunc={[Function]}
        name="description"
        placeholder="Describe your schnoodle"
        resize={false}
        rows={5}
        title="Description: " />
      <TextField
        content=""
        controlFunc={[Function]}
        inputType="text"
        name="url"
        placeholder="Put the link to your picture here"
        title="URL: " />
    </div>
    <button
      className="action-button shadow animate green"
      onClick={[Function]}>
      Submit
    </button>
    <button
      className="action-button shadow animate red"
      onClick={[Function]}>
      Clear Form
    </button>
  </fieldset>
</form>
`;

exports[`gallery view gallery display 1`] = `<ul />`;

exports[`list view list display 1`] = `<ul />`;

exports[`my buttons click handlers runs my test! 1`] = `
<div>
  <ImageView
    onClick={[Function]}
    view={null} />
</div>
`;

exports[`rendering list displays list 1`] = `
<div>
  <text
    className="view">
    Click a button to view schnoodles:
  </text>
  <ViewSelector />
</div>
`;

exports[`thumbnail view thumbnail display 1`] = `<ul />`;
